modeshape-jcr/src/main/java/org/modeshape/jcr/NodeTypes.java

```diff
@@ -238,6 +240,10 @@ protected NodeTypes( ExecutionContext context,
                     nodeTypeNamesThatAreShareable.add(name);
                 }
 
+                if (nodeType.isNodeType(ModeShapeLexicon.UNORDERED_COLLECTION)) {
+                    nodeTypeNamesThatAreUnorderableCollections.add(name);
```

Comment:
by virtue of the node type definitions: https://github.com/hchiorean/modeshape/blob/MODE-2109/modeshape-jcr/src/main/resources/org/modeshape/jcr/modeshape_builtins.cnd#L152
Any of the default and custom node types which are unordered collections are parsed when the node types are loaded.